rpython/rlib/test/test_rfile.py

```diff
@@ -135,11 +137,19 @@ def test_open_buffering_full(self):
         def f():
             f = open(fname, 'w', 128)
             f.write('dupa\ndupb')
-            f2 = open(fname, 'r')
+            f2 = open(fname, 'r', 0)
             assert f2.read() == ''
             f.write('z' * 120)
+            if __pypy__:
+                # PyPy will flush all 129 characters in a single op
+                # causing the second read bellow to fail without this:
+                f.write('z')
+
+            # This read should get all of the flushed data, but not everything
+            # (because our writes are buffered)
             assert f2.read() != ''
             f.close()
+            # Closing the file should flush the unread bytes
             assert f2.read() != ''
             f2.close()
```

Comment:
Can these be rewritten to use a context manager instead of managing the open/close?

Reply:
I don't think we can manage both in a context manager, since context managers are "LIFO":
```python
with open(...) as f1:
  with open(...) as f2:
    # here we'd like to close f1, but can't until after f2 is closed
    # the last-most open is closed first
```

I could add a single `with`, but mixing paradigms might not be the best for readability

Reply:
I take that back, we can manually `f1.close()` before exiting the context